VRCFaceTracking.Core/Models/OscTarget.cs

```diff
@@ -9,18 +10,50 @@ public partial class OscTarget : ObservableObject, IOscTarget
     [ObservableProperty] private bool _isConnected;
 
     [ObservableProperty] [property: SavedSetting("OSCInPort", 9001)]
+
     private int _inPort;
 
     [ObservableProperty] [property: SavedSetting("OSCOutPort", 9000)]
+
     private int _outPort;
 
-    [ObservableProperty] 
+    [ObservableProperty]
     [NotifyPropertyChangedFor(nameof(InPort))] [NotifyPropertyChangedFor(nameof(OutPort))]
+
     [property: SavedSetting("OSCAddress", "127.0.0.1")]
     private string _destinationAddress;
-    
+
+    private const string DefaultAddress = "127.0.0.1";
+    private bool _isReverting = false;
+
     public OscTarget(ILocalSettingsService localSettingsService)
     {
         PropertyChanged += (_, _) => localSettingsService.Save(this);
     }
-}
\ No newline at end of file
+
+    /// Called after the DestinationAddress property has changed.
+    /// It validates the new address and reverts it to the default if invalid.
+    partial void OnDestinationAddressChanged(string oldValue, string newValue)
```

Comment:
Might be an idea to use `NotifyDataErrorInfo` for this instead of just reverting the val
https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/observableproperty#requesting-property-validation

Reply:
Might even be able to create a custom validationattribute to validate it's a valid IP and be rid of that whole method

Reply:
Will look into it. Seems promising would also like to not have an validating function in the class if its not necessary.

Reply:
> Might be an idea to use `NotifyDataErrorInfo` for this instead of just reverting the val https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/observableproperty#requesting-property-validation

this is not directly possible since NotifyDataErrorInfo is an inheritance from ObservableProperty while the OSCTarget class is an child of ObservableObject which does not support it.

Reply:
Wat? The parent of OscTarget has no relevance. You should just be able to append the attribute below the existing ObservableProperty attribute of DestinationAddress

Reply:
Idk man it didn't allow me to do it this way, i just ended up doing your second suggestion which is a more elegant solution anyway in my opinion.